pkg/reconciler/taskrun/resources/taskref.go

```diff
@@ -181,43 +180,48 @@ func resolveTask(ctx context.Context, resolver remote.Resolver, name string, kin
 }
 
 // readRuntimeObjectAsTask tries to convert a generic runtime.Object
-// into a v1beta1.TaskObject type so that its meta and spec fields
+// into a *v1.Task type so that its meta and spec fields
 // can be read. v1 object will be converted to v1beta1 and returned.
 // An error is returned if the given object is not a
 // TaskObject or if there is an error validating or upgrading an
 // older TaskObject into its v1beta1 equivalent.
 // TODO(#5541): convert v1beta1 obj to v1 once we use v1 as the stored version
-func readRuntimeObjectAsTask(ctx context.Context, obj runtime.Object) (v1beta1.TaskObject, error) {
+func readRuntimeObjectAsTask(ctx context.Context, obj runtime.Object) (*v1.Task, error) {
```

Comment:
Currently I have tried replacing v1beta1.TaskObject with v1.Task, but not sure if there are any concerns here.

Reply:
won't this break clusterTasks?

Reply:
I might be mistaken, I thought during the resolveTask process we could convert v1beta1.ClustertASK to  a v1.Task

Reply:
oh sorry, I didn't look closely enough. I think this could work, but I suggest making this change in its own PR, and testing it out locally.

Reply:
Thanks for the pointer. Going to try to separate `v1beta1.TaskObject` out for `v1beta1.Task` and `ClusterTask` in a single PR.